lib/vtls/openssl.h

```diff
@@ -49,8 +49,16 @@
 #define HAVE_KEYLOG_CALLBACK
 #endif
 
+/* Check for OpenSSL 1.1.1 which has early data support. */
+#undef HAVE_OPENSSL_EARLYDATA
+#if OPENSSL_VERSION_NUMBER >= 0x10100010L && defined(TLS1_3_VERSION) && \
+    !defined(OPENSSL_IS_BORINGSSL) && !defined(OPENSSL_IS_AWSLC)
```

Comment:
```suggestion
#if defined(TLS1_3_VERSION) && \
    ((OPENSSL_VERSION_NUMBER >= 0x10100010L && \
      !defined(LIBRESSL_VERSION_NUMBER)) || \
     (defined(LIBRESSL_VERSION_NUMBER) && \
      LIBRESSL_VERSION_NUMBER >= 0x3040100fL)) && \
    !defined(OPENSSL_IS_BORINGSSL) && !defined(OPENSSL_IS_AWSLC)
```
Enable for LibreSSL? It seems supported since 3.4.1 (2021-10-14):
https://ftp.openbsd.org/pub/OpenBSD/LibreSSL/libressl-3.4.1-relnotes.txt

Reply:
@vszakats thanks for the link. I just checked with libreSSL v3.9.2 and they have the API, but there is no implementation behind it. For example, their code for asking a session about the max early data the server  announced is:

```c
uint32_t
SSL_SESSION_get_max_early_data(const SSL_SESSION *s)
{
	return 0;
}
```

Hmm, yeah. Let's say this is not a supported feature.